### ISSUE TITLE
[Bug]: Annotations management in record (CtRecordImpl) may raise NPE when annotation has no Target (meaning ALL)

### ISSUE THREAD

**Issue by guillaumem-theodo** [INRIA/spoon#5102]
### Describe the bug

When a record has fields annotated with Annotation (example @Nullable from google.jsr OR custom annotation)  without @Target the CtRecordImpl raises an NullPointerException (line 135). 

Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.annotation.Target.value()" because "target" is null
	at spoon.support.reflect.declaration.CtRecordImpl.getAnnotationsWithName(CtRecordImpl.java:135)
	at spoon.support.reflect.declaration.CtRecordImpl.addTypeMemberAt(CtRecordImpl.java:108)

### Source code you are trying to analyze/transform

```Java
@Retention(RetentionPolicy.RUNTIME)
public @interface MyAnnotation {
}


and 

public record ExampleRecord(String a, @MyAnnotation  String b) {
}
```

### Source code for your Spoon processing

```Java
MavenLauncher launcher = new MavenLauncher(".", MavenLauncher.SOURCE_TYPE.APP_SOURCE, true);
        launcher.getEnvironment().setNoClasspath(true);
        launcher.getEnvironment().setIgnoreDuplicateDeclarations(true);
        CtModel ctModel = launcher.buildModel();
```

### Actual output

```Java
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.lang.annotation.Target.value()" because "target" is null
	at spoon.support.reflect.declaration.CtRecordImpl.getAnnotationsWithName(CtRecordImpl.java:135)
	at spoon.support.reflect.declaration.CtRecordImpl.addTypeMemberAt(CtRecordImpl.java:108)
```

### Expected output

```Java
No NPE
```

### Spoon Version

10.2.0

### JVM Version

16

### What operating system are you using?

MacOS 11.7

**Comment by guillaumem-theodo** [comment]
Quote from java @Target documentation.

If an @Target meta-annotation is not present on an annotation interface T, then an annotation of type T may be written as a modifier for any declaration.

Thus an annotation may not have a @Target.

The code of `CtRecordImpl.java` should test the presence before iterating on values (in method getAnnotationsWithName). 

Replace:
`
					if (annotationType != null) {
						Target target = annotationType.getAnnotation(Target.class);
						if (Arrays.stream(target.value()).anyMatch(e -> e == elementType)) {
							result.add(annotation.clone());
						}
					}
`

By:
`
					if (annotationType != null) {
						Target target = annotationType.getAnnotation(Target.class);
						if (target == null || Arrays.stream(target.value()).anyMatch(e -> e == elementType)) {
							result.add(annotation.clone());
						}
					}
`

**Comment by SirYwell** [comment]
This is already fixed: https://github.com/INRIA/spoon/blob/5ff045d34b1f1312e4e059489bb93968cb99e889/src/main/java/spoon/support/reflect/declaration/CtRecordImpl.java#L142

but the fix was after the 10.2.0 release (https://github.com/INRIA/spoon/pull/5016). Could you try with `10.3.0-beta-14`? (I'm not sure when this version was released, so I don't know if the fix is included there already)

**Comment by guillaumem-theodo** [comment]
Thanks for your quick answer. I confirm the bug is fixed in 10.3.0-beta-14 version. Thanks a lot.

**Comment by I-Al-Istannen** [comment]
Thanks for reporting, I think this issue can be closed now. Feel free to reopen if not :)
